storage/storage-interface/src/state_store/state_view/cached_state_view.rs

```diff
@@ -151,150 +99,118 @@ impl CachedStateView {
-                });
+        rayon::scope(|s| {
+            keys.into_iter().for_each(|key| {
+                s.spawn(move |_| {
+                    self.get_state_value(key).expect("Must succeed.");
+                })
+            });
         });
         Ok(())
     }
 
-    pub fn into_state_cache(self) -> StateCache {
-        StateCache {
-            frozen_base: self.speculative_state,
-            sharded_state_cache: self.sharded_state_cache,
-            proofs: self.proof_fetcher.get_proof_cache(),
-        }
+    /// Consumes `Self` and returns the state and all the memorized state reads.
+    pub fn into_memorized_reads(self) -> ShardedStateCache {
+        let Self {
+            id: _,
+            reader: _,
+            speculative: _,
+            memorized,
+        } = self;
+
+        memorized
     }
 
-    fn get_version_and_state_value_internal(
-        &self,
-        state_key: &StateKey,
-    ) -> Result<(Option<Version>, Option<StateValue>)> {
-        // Do most of the work outside the write lock.
-        let key_hash = state_key.hash();
-        match self.speculative_state.get(key_hash) {
-            StateStoreStatus::ExistsInScratchPad(value) => Ok((None, Some(value))),
-            StateStoreStatus::DoesNotExist => Ok((None, None)),
-            // Part of the tree is unknown, need to request proof for later usage (updating the tree)
-            StateStoreStatus::UnknownSubtreeRoot {
-                hash: subtree_root_hash,
-                depth: subtree_root_depth,
-            } => self.fetch_value_and_maybe_proof_in_snapshot(
-                state_key,
-                Some((subtree_root_hash, subtree_root_depth)),
-            ),
-            // Tree is known, but we only know the hash of the value, need to request the actual
-            // StateValue.
-            StateStoreStatus::UnknownValue => {
-                self.fetch_value_and_maybe_proof_in_snapshot(state_key, None)
-            },
-        }
+    fn base_version(&self) -> Option<Version> {
+        self.speculative.base_version()
     }
 
-    fn fetch_value_and_maybe_proof_in_snapshot(
-        &self,
-        state_key: &StateKey,
-        fetch_proof: Option<(HashValue, usize)>,
-    ) -> Result<(Option<Version>, Option<StateValue>)> {
-        let version_and_value_opt = match self.snapshot {
-            None => None,
-            Some((version, _root_hash)) => match fetch_proof {
-                None => self.proof_fetcher.fetch_state_value(state_key, version)?,
-                Some((subtree_root_hash, subtree_depth)) => self
-                    .proof_fetcher
-                    .fetch_state_value_with_version_and_schedule_proof_read(
-                        state_key,
-                        version,
-                        subtree_depth,
-                        Some(subtree_root_hash),
-                    )?,
-            },
+    fn get_uncached(&self, state_key: &StateKey) -> Result<StateCacheEntry> {
```

Comment:
why do we choose not to pre-fetch proof ?

Reply:
At execution stage we don't know which keys are actually written to (versus only read), that's a waste of CPU, same as today.
And with this change, we don't have the tree at the parent version, so we can't fetch partial proofs, which is added waste of CPU. 

We can get around these with some (pretty complicated) tricks, but with our long-term goals (less updates to the global tree) in mind, I'd rather keep it simple as long as it's not too big of a latency concern.

Reply:
sgtm. can we test if the latency or TPS is impacted?

Reply:
Throughput is an improvement in all cases now. latency I didn't see much change from forge.

Also see the summary.